src/ServiceControl/Infrastructure/Settings/Settings.cs

```diff
@@ -180,6 +181,10 @@ public TimeSpan HeartbeatGracePeriod
 
         public bool DisableHealthChecks { get; set; }
 
+        // The default value is set to the maximum allowed time by the most restrictive
+        // hosting platform, which Docker Linux containers. 
+        public TimeSpan ShutdownTimeout { get; set; } = TimeSpan.FromSeconds(10);
```

Comment:
10 seconds is when we get killed, you set the timeout to allow for cancellation.

As in:

- if we're interested in cancellation, set it to less than the host its kill duration
- if we're not interested in cancellation, set this to a high value that gives us time to run a graceful termination

With external, I think 5 seconds is already plenty
With internal, I want to wait as long as possible

So I like your last commits, but if we want to see cancellation to take place this number should be even lower:

```suggestion
        // The default value is set to the maximum allowed time by the most restrictive
        // hosting platform, which Docker Linux containers minus a few seconds
        // to actually allow for cancellation and logging to take place
        public TimeSpan ShutdownTimeout { get; set; } = TimeSpan.FromSeconds(5);
```